### PR TITLE
feat(serving): add option to read xref pages concurrently

### PR DESCRIPTION
**Pull request by schroederc**

### CODE REVIEW

**Comment by shahms** [comment]
```
go: updates to go.mod needed, disabled by -mod=readonly; to update it:
	go mod tidy
```

File: kythe/go/serving/xrefs/xrefs.go [hunk 1]
```diff
@@ -624,6 +627,53 @@ func decorationToReference(norm *span.Normalizer, d *srvpb.FileDecorations_Decor
...
+			reply.Total.Declarations += int64(idx.Count)
+		} else {
+			reply.Filtered.Declarations += int64(idx.Count)
+		}
+	case xrefCategoryRef:
+		if pageSet.Contains(idx) {
+			reply.Total.References += int64(idx.Count)
+		} else {
+			reply.Filtered.References += int64(idx.Count)
+		}
```
Comment (shahms) [inline kythe/go/serving/xrefs/xrefs.go:661]: should this be?
```
case xrefCategoryDef, xrefCategoryDecl, xrefCategoryRef:
		if pageSet.Contains(idx) {
			reply.Total.Definitions += int64(idx.Count)
		} else {
			reply.Filtered.Definitions += int64(idx.Count)
		}
```

File: kythe/go/serving/xrefs/xrefs.go [hunk 2]
```diff
@@ -916,12 +1031,7 @@ readLoop:
 					reply.Filtered.Declarations += int64(filtered)
 					stats.addAnchors(&crs.Declaration, p.Group)
 				}
-			case xrefs.IsRefKind(req.ReferenceKind, idx.Kind):
-				if !pageSet.Contains(idx) {
-					reply.Filtered.References += int64(idx.Count)
-					continue
-				}
-				reply.Total.References += int64(idx.Count)
+			case xrefCategoryRef:
```
Comment (shahms) [inline kythe/go/serving/xrefs/xrefs.go:1034]: same for these three. The case-code for xrefCategory{Def,Decl,Ref} appears identical; should they be a single case.

File: kythe/go/serving/xrefs/xrefs.go (hunk 1 above)
Comment (schroederc) [reply kythe/go/serving/xrefs/xrefs.go:661]: I don't understand.  Why would decls and refs both be counted as defs?

File: kythe/go/serving/xrefs/xrefs.go (hunk 2 above)
Comment (schroederc) [reply kythe/go/serving/xrefs/xrefs.go:1034]: They affect different categories in the reply.

**Comment by shahms** [comment]
Apparently my eyes don't work today.

**Pull request merged**